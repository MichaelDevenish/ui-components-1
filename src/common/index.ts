export { Props } from './types'
export { getColor } from './legacy'
