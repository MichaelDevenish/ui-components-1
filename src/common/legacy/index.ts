export { getColor } from './colors/getColor'
