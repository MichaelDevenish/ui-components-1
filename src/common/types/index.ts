/**
 * For Examples, see:
 * https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts
 * https://github.com/alm-tools/alm/blob/master/src/common/types.ts
 */
import { Size as _s, AllSizes } from './size'

namespace Props {
  export import Size = _s
}

export {
  Props,
  AllSizes
}
