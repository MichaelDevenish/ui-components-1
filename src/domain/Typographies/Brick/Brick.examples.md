#### Styling of Bricks

```jsx
<div>
  <Brick
    text='Alert!!!'
    color='alert'
  />

  <Brick
    text='Success'
    color='success'
  />

  <Brick
    text='Warning!'
    color='warning'
  />

  <Brick
    text='Primary'
    color='primary'
  />

  <Brick
    text='Neutral (default)'
    color='neutral'
  />

  <Brick
    text='Secondary'
    color='secondary'
  />

  <Brick
    text='Highlight'
    color='highlight'
  />
  <Brick
    text='Dark'
    color='dark'
  />
</div>
```

#### Bricks with Typographies

```jsx
<div>
  <div>
    <Brick
      text='X-Small'
      color='dark'
      typographyType='xsmall'
    />
  </div>
  <div>
    <Brick
      text='Small'
      color='dark'
      typographyType='small'
    />
  </div>
  <div>
    <Brick
      text='Body (default)'
      color='dark'
      typographyType='body'
    />
  </div>
  <div>
    <Brick
      text='Heading'
      color='dark'
      typographyType='heading'
    />
  </div>
  <div>
    <Brick
      text='Display'
      color='dark'
      typographyType='display'
    />
  </div>
  <div>
    <Brick
      text='Display Large'
      color='dark'
      typographyType='display-large'
    />
  </div>
</div>
```
