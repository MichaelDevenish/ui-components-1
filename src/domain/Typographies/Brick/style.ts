import styled, { css } from 'styled-components'
import { Props, Variables } from '../../../common'
import { styleForTypographyType } from '../services/textStyles'

interface IBrickWrapperProps {
  /** Color of the brick  */
  color: BrickColor
  /** Specify the type of typography to use */
  typographyType: Props.TypographyType
}

enum BrickColor {
  Alert = 'alert',
  Success = 'success',
  Warning = 'warning',
  Primary = 'primary',
  Neutral = 'neutral',
  Secondary = 'secondary',
  Highlight = 'highlight',
  Dark = 'dark'
}

const colors = {
  [BrickColor.Alert]: {
    textColor: Variables.Color.r600,
    backgroundColor: Variables.Color.r100
  },
  [BrickColor.Success]: {
    textColor: Variables.Color.g600,
    backgroundColor: Variables.Color.g100
  },
  [BrickColor.Warning]: {
    textColor: Variables.Color.o600,
    backgroundColor: Variables.Color.o100
  },
  [BrickColor.Primary]: {
    textColor: Variables.Color.i600,
    backgroundColor: Variables.Color.i100
  },
  [BrickColor.Neutral]: {
    textColor: Variables.Color.n800,
    backgroundColor: Variables.Color.n250
  },
  [BrickColor.Secondary]: {
    textColor: Variables.Color.b600,
    backgroundColor: Variables.Color.b100
  },
  [BrickColor.Highlight]: {
    textColor: Variables.Color.c600,
    backgroundColor: Variables.Color.c100
  },
  [BrickColor.Dark]: {
    textColor: Variables.Color.n200,
    backgroundColor: Variables.Color.n700
  }
}

const BrickWrapper = styled.span<IBrickWrapperProps>`
  border-radius: ${Variables.Style.borderRadius}px;
  padding: 2px 4px;
  word-break: break-word;

  ${(props: IBrickWrapperProps) => styleForTypographyType(props.typographyType)}
  ${(props: IBrickWrapperProps) => {
    const color = colors[props.color]
    return css `
      background: ${color.backgroundColor};
      color: ${color.textColor};
    `
  }}
`

export {
  BrickColor,
  BrickWrapper,
  IBrickWrapperProps
}
