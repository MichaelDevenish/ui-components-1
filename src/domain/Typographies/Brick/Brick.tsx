import React from 'react'
import { Props, Variables } from '../../../common'
import { BrickWrapper } from './style'
import { Text } from '../Text'

export interface IBrickProps {
  /** Text to show inside the Brick  */
  text: string
  /** Color of the brick  */
  color?: Props.FunctionalColor
  /** Specify the type of typography to use */
  typographyType?: Props.TypographyType
  /** Extra classes to apply */
  className?: string
  /** The data-component-context */
  componentContext?: string
}

export class Brick extends React.PureComponent<IBrickProps> {
  public static defaultProps: Partial<IBrickProps> = {
    color: Props.FunctionalColor.Neutral,
    typographyType: Props.TypographyType.Body
  }

  public render (): JSX.Element {
    const {
      text,
      color,
      typographyType,
      className,
      componentContext
    } = this.props

    const brickColors = Variables.functionalColors[color!]

    return (
      <BrickWrapper
        color={brickColors.backgroundColor}
        className={className}
        data-component-type={Props.ComponentType.Brick}
        data-component-context={componentContext}
      >
        <Text
          type={typographyType}
          color={brickColors.textColor}
        >
          {text}
        </Text>
      </BrickWrapper>
    )
  }
}
