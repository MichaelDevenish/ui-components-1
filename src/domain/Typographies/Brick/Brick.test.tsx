import { shallow } from 'enzyme'
import React from 'react'

import { Props } from '../../../common'
import { Brick } from './Brick'

describe('<Brick />', () => {
  it('should render a Brick with default values', () => {
    const wrapper = shallow(
      <Brick
      >
        Test
      </Brick>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render a Brick with the alert color', () => {
    const wrapper = shallow(
      <Brick
        color={Brick.Color.Alert}
      >
        Test
      </Brick>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render a Brick with the heading typography', () => {
    const wrapper = shallow(
      <Brick
        typographyType={Props.TypographyType.Heading}
      >
        Test
      </Brick>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
