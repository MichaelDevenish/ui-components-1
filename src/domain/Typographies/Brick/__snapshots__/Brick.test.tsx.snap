// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brick /> should render a Brick with default values 1`] = `
<styled.span
  color="neutral"
  data-component-type="brick"
  typographyType="body"
>
  Test
</styled.span>
`;

exports[`<Brick /> should render a Brick with the alert color 1`] = `
<styled.span
  color="alert"
  data-component-type="brick"
  typographyType="body"
>
  Test
</styled.span>
`;

exports[`<Brick /> should render a Brick with the heading typography 1`] = `
<styled.span
  color="neutral"
  data-component-type="brick"
  typographyType="heading"
>
  Test
</styled.span>
`;
